client/blockchain-service/src/handler.rs

```diff
@@ -741,4 +796,61 @@ impl BlockchainService {
             return Err(anyhow::anyhow!("Failed to get Events storage element"));
         }
     }
+
+    /// Check if the challenges tick is one that this provider has to submit a proof for,
+    /// and if so, emit a `NewChallengeSeed` event.
+    fn should_provider_submit_proof(
+        &self,
+        block_hash: &H256,
+        provider_id: &ProviderId,
+        current_tick: &BlockNumber,
+    ) -> bool {
+        let last_tick_provided = match self
+            .client
+            .runtime_api()
+            .get_last_tick_provider_submitted_proof(block_hash.clone(), provider_id)
+        {
+            Ok(last_tick_provided_result) => match last_tick_provided_result {
+                Ok(last_tick_provided) => last_tick_provided,
+                Err(e) => match e {
+                    GetLastTickProviderSubmittedProofError::ProviderNotRegistered => {
+                        debug!(target: LOG_TARGET, "Provider [{:?}] is not registered", provider_id);
+                        return false;
+                    }
+                    GetLastTickProviderSubmittedProofError::ProviderNeverSubmittedProof => {
+                        debug!(target: LOG_TARGET, "Provider [{:?}] does not have an initialised challenge cycle", provider_id);
+                        return false;
+                    }
+                },
+            },
+            Err(e) => {
+                error!(target: LOG_TARGET, "Runtime API error while getting last tick Provider [{:?}] submitted a proof for: {:?}", provider_id, e);
+                return false;
+            }
+        };
+        let provider_challenge_period = match self
+            .client
+            .runtime_api()
+            .get_challenge_period(block_hash.clone(), provider_id)
+        {
+            Ok(provider_challenge_period_result) => match provider_challenge_period_result {
+                Ok(provider_challenge_period) => provider_challenge_period,
+                Err(e) => match e {
+                    GetChallengePeriodError::ProviderNotRegistered => {
+                        debug!(target: LOG_TARGET, "Provider [{:?}] is not registered", provider_id);
+                        return false;
+                    }
+                },
+            },
+            Err(e) => {
+                debug!(target: LOG_TARGET, "Runtime API error while getting challenge period for Provider [{:?}]: {:?}", provider_id, e);
+                return false;
+            }
+        };
+
+        trace!(target: LOG_TARGET, "CURRENT TICK: {:?}", current_tick);
+        trace!(target: LOG_TARGET, "LAST TICK PROVED: {:?}", last_tick_provided);
+        trace!(target: LOG_TARGET, "PROVIDER CHALLENGE PERIOD: {:?}", provider_challenge_period);
```

Comment:
Maybe it would be better to print these in the same `trace!` call to avoid other logs to show up in the middle in some cases?